swarm_copy/bluenaas_models.py

```diff
@@ -0,0 +1,307 @@
+    updated_at: Annotated[datetime, Field(title="Updated At")]
+
+
+class ExclusionRule(BaseModel):
+    """Placeholder."""
+
+    distance_soma_gte: Annotated[Optional[float], Field(title="Distance Soma Gte")] = (
+        None
+    )
+    distance_soma_lte: Annotated[Optional[float], Field(title="Distance Soma Lte")] = (
+        None
+    )
+
+
+class ExperimentSetupConfig(BaseModel):
+    """Placeholder."""
+
+    celsius: Annotated[float, Field(title="Celsius")]
+    vinit: Annotated[float, Field(title="Vinit")]
+    hypamp: Annotated[float, Field(title="Hypamp")]
+    max_time: Annotated[float, Field(le=3000.0, title="Max Time")]
+    time_step: Annotated[float, Field(title="Time Step")]
+    seed: Annotated[int, Field(title="Seed")]
+
+
+class RecordingLocation(BaseModel):
+    """Placeholder."""
+
+    section: Annotated[str, Field(title="Section")]
+    offset: Annotated[float, Field(ge=0.0, le=1.0, title="Offset")]
+
+
+class SectionTarget(RootModel[Literal["apic", "basal", "dend", "soma", "axon"]]):
+    """Placeholder."""
+
+    root: Annotated[
+        Literal["apic", "basal", "dend", "soma", "axon"], Field(title="SectionTarget")
+    ]
+
+
+class SimulationStimulusConfig(BaseModel):
+    """Placeholder."""
+
+    stimulus_type: Annotated[
+        Literal["current_clamp", "voltage_clamp", "conductance"],
+        Field(title="Stimulus Type"),
+    ]
+    stimulus_protocol: Annotated[
+        Optional[Literal["ap_waveform", "idrest", "iv", "fire_pattern"]],
+        Field(title="Stimulus Protocol"),
+    ] = None
+    amplitudes: Annotated[Union[List[float], float], Field(title="Amplitudes")]
+
+
+class StimulationItemResponse(BaseModel):
+    """Placeholder."""
+
+    x: Annotated[List[float], Field(title="X")]
+    y: Annotated[List[float], Field(title="Y")]
+    name: Annotated[str, Field(title="Name")]
+    amplitude: Annotated[float, Field(title="Amplitude")]
+
+
+class StimulationPlotConfig(BaseModel):
+    """Placeholder."""
+
+    stimulus_protocol: Annotated[
+        Optional[Literal["ap_waveform", "idrest", "iv", "fire_pattern"]],
+        Field(title="Stimulus Protocol"),
+    ] = None
+    amplitudes: Annotated[List[float], Field(title="Amplitudes")]
+
+
+class SynapseConfig(BaseModel):
+    """Placeholder."""
+
+    id: Annotated[str, Field(title="Id")]
+    name: Annotated[str, Field(title="Name")]
+    target: Optional[SectionTarget] = None
+    type: Annotated[int, Field(title="Type")]
+    distribution: Annotated[
+        Literal["exponential", "linear", "formula"], Field(title="Distribution")
+    ]
+    formula: Annotated[Optional[str], Field(title="Formula")] = None
+    soma_synapse_count: Annotated[Optional[int], Field(title="Soma Synapse Count")] = (
+        None
+    )
+    seed: Annotated[int, Field(title="Seed")]
+    exclusion_rules: Annotated[
+        Optional[List[ExclusionRule]], Field(title="Exclusion Rules")
+    ] = None
+
+
+class SynapsePlacementBody(BaseModel):
```

Comment:
So you want to add synapse generation too in this mr ?

Reply:
No, this is from the bluenaas api. It's not used for now.